streaming-pubsub/src/main/scala/org/apache/spark/streaming/pubsub/PubsubInputDStream.scala

```diff
@@ -262,37 +283,32 @@ class PubsubReceiver(
         }
       case None => // do nothing
     }
+
     new Thread() {
       override def run() {
         receive()
       }
     }.start()
+
+    // Scheduling update rate limit method at every second.
+    scheduledExecutor.scheduleAtFixedRate(
+      new Runnable {
+        override def run(): Unit = updateRateLimit()
```

Comment:
Do we really need to call it in a separate thread? Can we "updateRateLimit" after every pull requests? Any downside to updateRateLimit in same thread?